app/src/main/java/com/arctouch/codechallenge/data/MoviesDataSource.java

```diff
@@ -0,0 +1,33 @@
+package com.arctouch.codechallenge.data;
+
+import androidx.paging.PageKeyedDataSource;
+
+import com.arctouch.codechallenge.data.model.Genre;
+import com.arctouch.codechallenge.data.model.GenreResponse;
+import com.arctouch.codechallenge.data.model.Movie;
+import com.arctouch.codechallenge.data.model.UpcomingMoviesResponse;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.reactivex.Observable;
+
+public abstract class MoviesDataSource /*extends PageKeyedDataSource<Integer, Movie>*/ {
+    //public abstract Observable<UpcomingMoviesResponse> subscribeForAllmovies(String language, int page, String region);
```

Comment:
What do you think of leaving comments and unused methods/fields in the code?

Reply:
It is not good at all, doing so could cause your classes to be difficult to read. Sorry for that.